src/store/preferences.ts

```diff
@@ -118,17 +127,22 @@ export const actions = {
    * @param args Key, value pair that corresponds to a property and its value
    * in the preferences object
    */
-  updatePreferencesData<P extends RecursiveKeys<Settings>>({
+  async updatePreferencesData<P extends RecursiveKeys<Settings>>({
     commit, dispatch, state, rootState,
-  }: PrefActionContext, args: {property: P, value: RecursiveTypes<Settings>[P]}): void {
+  }: PrefActionContext, args: {property: P, value: RecursiveTypes<Settings>[P]}): Promise<void> {
     const { property, value } = args;
 
-    commit('SET_PREFERENCES', _.set(_.cloneDeep(state.preferences), property, value));
-    dispatch(
+    const newPreferences = _.set(_.cloneDeep(state.preferences), property, value);
+
+    await dispatch(
```

Comment:
This dispatch is for the side effect of setting severities? Does that need a comment?

Reply:
I updated the docstring for `proposePreferences` to call this out.